tests-functional/clients/status_backend.py

```diff
@@ -4,23 +4,32 @@
 import random
 import threading
 import requests
-from tenacity import retry, stop_after_delay, wait_fixed
+import docker
+import os
 
+from tenacity import retry, stop_after_delay, wait_fixed
 from clients.signals import SignalClient
 from clients.rpc import RpcClient
 from datetime import datetime
 from conftest import option
-from constants import user_1, DEFAULT_DISPLAY_NAME
+from constants import user_1, DEFAULT_DISPLAY_NAME, USER_DIR
 
 
 class StatusBackend(RpcClient, SignalClient):
 
-    def __init__(self, await_signals=[], url=None):
-        try:
-            url = url if url else random.choice(option.status_backend_urls)
-        except IndexError:
-            raise Exception("Not enough status-backend containers, please add more")
-        option.status_backend_urls.remove(url)
+    def __init__(self, await_signals=[], new_container=True):
+
+        if new_container:
+
+            self.docker_client = docker.from_env()
+            host_port = random.choice(option.status_backend_port_range)
+
+            self.container = self._start_container(host_port)
+            url = f"http://0.0.0.0:{host_port}"
```

Comment:
Should be localhost:
```suggestion
            url = f"http://127.0.0.1:{host_port}"
```